### ISSUE TITLE
[Bug]: Terraform doesn't allow data larger than 16K for aws_imagebuilder_component

### ISSUE THREAD

**Issue by Casey-Bateman** [hashicorp/terraform-provider-aws#29377]
### Terraform Core Version

1.3.5

### AWS Provider Version

4.41.0

### Affected Resource(s)

* aws_imagebuilder_component

### Expected Behavior

When updating an `aws_imagebuilder_component` with a `data` field larger than 16KB it should be marked as valid based on the service quota. At the very least `data` would be extended to 64KB as that is the default service limit for a component size. 

### Actual Behavior

The process failed with an error:
```
Error: expected length of data to be in the range (1-16000), got <full document>
...
```

### Relevant Error/Panic Output Snippet

```shell
Error: expected length of data to be in the range (1-16000), got <full document>
...
```

### Terraform Configuration Files

No response

### Steps to Reproduce

Create an `aws_imagebuilder_component` with a data field that is larger than 16K. Can be done in terraform by specifying a `file(...)` as input. example:
```
resource "aws_imagebuilder_component" "really_big_component" {
  name     = "my_test_really_big_component"
  platform = "Linux"
  version  = "0.1.0"
  data     = file("./really_big_component.yaml")
}
```

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @Casey-Bateman 👋 Thank you for taking the time to raise this! It looks like this is a [limit imposed by the AWS API](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_CreateComponent.html#API_CreateComponent_RequestSyntax). Are you seeing different results elsewhere that let you to believing that this should be possible via the API? If so, we could investigate whether the linked document is out of date, and adjust the provider accordingly.

**Comment by Casey-Bateman** [comment]
@justinretzolk In working with our AWS Reps they had said that the limit should be the default Container Size limit of 64KB. I will work with our reps again and let them know that they are enforcing the limit through the API that you mentioned. I will report back with any updates from them.

**Comment by Casey-Bateman** [comment]
I was just informed by our AWS Reps that a Product Feature request has been created to raise that limit. This will not be a relatively quick turn around. So if preferred, we can close this issue and re-visit when the AWS limit has been raised?

**Comment by justinretzolk** [comment]
Given that this is upstream and is potentially being worked on, I think it's probably better to close this one out and then open a new issue once feature work is complete on the AWS side. Thanks for coming back and letting us know, and apologies that it took so long for me to get back to you!
